api/app/signals/apps/api/generics/relations.py

```diff
@@ -0,0 +1,38 @@
+from rest_framework import relations
+from rest_framework.reverse import reverse
+
+
+class ParameterisedHyperlinkedRelatedField(relations.HyperlinkedRelatedField):
+    lookup_fields = (('pk', 'pk'),)
+
+    def __init__(self, *args, **kwargs):
+        self.lookup_fields = kwargs.pop('lookup_fields', self.lookup_fields)
+        super(ParameterisedHyperlinkedRelatedField, self).__init__(*args, **kwargs)
+
+    def use_pk_only_optimization(self):
+        return False
+
+    def get_object(self, view_name, view_args, view_kwargs):
+        kwargs = {}
+        for model_field, url_param in self.lookup_fields:
+            model_field = model_field.replace('.', '__')
+            kwargs[model_field] = view_kwargs[url_param]
+        return self.get_queryset().get(**kwargs)
+
+    def get_url(self, obj, view_name, request, format):
+        kwargs = {}
+        for model_field, url_param in self.lookup_fields:
+            attr = obj
```

Comment:
This is confusing, you trying to get a named attribute from some object using `getattr` below. Why not leave it as `obj` (which is also the name normally used in DRF)?

Reply:
Because we also need to follow relations. So I wanted to make it obvious we are using the attr from the obj

Reply:
It is only a naming thing, consider using another temporary variable name.